fido/Crypto/Fido2/Certificate.hs

```diff
@@ -0,0 +1,56 @@
+-- | Utility functions for handling the certificates used during attestation
+module Crypto.Fido2.Certificate (hasDnElement, findDnElement, findProperExtension, decodeAAGUID, fieldEquals) where
+
+import Crypto.Fido2.Error (AttestationError (ASN1Error, AttestationCredentialAAGUIDMissing))
+import Data.ASN1.BinaryEncoding (DER (DER))
+import Data.ASN1.Encoding (ASN1Decoding (decodeASN1))
+import qualified Data.ASN1.OID as OID
+import Data.ASN1.Parse (ParseASN1, getMany, getNext, getNextContainerMaybe, hasNext, runParseASN1)
+import Data.ASN1.Types (ASN1 (OctetString), ASN1Class (Context), ASN1ConstructionType (Container), ASN1Tag)
+import Data.ByteString (ByteString)
+import Data.ByteString.Lazy (fromStrict)
+import Data.Either (fromRight)
+import Data.List (find)
+import Data.Maybe (isJust)
+import qualified Data.X509 as X509
+
+hasDnElement :: X509.DnElement -> [(OID.OID, X509.ASN1CharacterString)] -> Bool
+hasDnElement el = isJust . findDnElement el
+
+findDnElement :: X509.DnElement -> [(OID.OID, X509.ASN1CharacterString)] -> Maybe X509.ASN1CharacterString
+findDnElement dnElementName = lookup (OID.getObjectID dnElementName)
+
+findProperExtension :: OID.OID -> [X509.ExtensionRaw] -> Maybe X509.ExtensionRaw
+findProperExtension extensionOID = find ((==) extensionOID . X509.extRawOID)
+
+decodeAAGUID :: ByteString -> Either AttestationError ByteString
+decodeAAGUID bs = do
+  asn1 <- either (Left . ASN1Error) pure . decodeASN1 DER $ fromStrict bs
+  case asn1 of
+    [OctetString s] -> Right s
+    _ -> Left AttestationCredentialAAGUIDMissing
+
+-- | Consumes ASN elements untill it finds the resulting field, then checks the value of that field
+--   Currently works for https://source.android.com/security/keystore/attestation
+-- TODO: This function can be implemented alternatively by not using the
+-- getNextContainerMaybe function and looking for the start of the function ourself
+fieldEquals :: ASN1Tag -> [ASN1] -> [ASN1] -> Bool
+fieldEquals tag val = fromRight False . runParseASN1 findTag
```

Comment:
This function also won't be necessary anymore with `AuthorizationList` type.

Also note that the move of these functions to this new `Certificate.hs` module isn't necessary anymore, they could stay where they were before.